service/controller/v2/resource/chart/resource.go

```diff
@@ -69,6 +71,21 @@ func (r *Resource) Name() string {
 	return Name
 }
 
+func isChartModified(a, b ChartState) bool {
+	// ChartValues have changed so we need to update.
+	if !reflect.DeepEqual(a.ChartValues, b.ChartValues) {
+		return true
+	}
+
+	// ReleaseVersion has changed for the channel so we need to update.
```

Comment:
Same as above, the comment assumes that the check is going to be true.

Reply:
Done.